### PR TITLE
feat: updated workflow with more skippable PRs

### LINKED ISSUE #200: Testing workflow could skip earlier
EDIT, only step 2 is in scope

1. I can't remember why we have [this if statement](https://github.com/asyncapi/.github/blob/master/.github/workflows/if-nodejs-pr-testing.yml#L19) on step and not job level 🤔 We could move it to job level IMHO and therefore workflow would never run - CLARIFIED - out of scope
2. go and nodejs testing workflow could be updated with more "cancel" conditions. In the case of PRs related to workflows updates, now there are more commit messages that we should consider as skippable -> https://github.com/asyncapi/.github/blob/master/.github/workflows/global-replicator.yml, not only `ci: update global workflows`

@KhudaDad414 what do you think about the 1st one?

### PR DESCRIPTION
**Pull request by dante381**
**Description**
-Added 8 other PR titles to each `if` statement in testing workflow.

**Related issue(s)**

Resolves #200

### CODE REVIEW

File: .github/workflows/if-docker-pr-testing.yml [hunk 1]
```diff
@@ -16,7 +16,24 @@ jobs:
...
+              startsWith(github.event.pull_request.title, 'ci: update release-related workflows for nodejs projects') || 
+              startsWith(github.event.pull_request.title, 'ci: update semantic release config file') || 
+              startsWith(github.event.pull_request.title, 'ci: update generic workflows') || 
+              startsWith(github.event.pull_request.title, 'ci: update workflows for docker-based projects') ||
+              startsWith(github.event.pull_request.title, 'chore(release):')
+            )) ||
+            (github.actor == 'allcontributors' && 
+              startsWith(github.event.pull_request.title, 'docs: add')
+            ))
+          )"
```
Comment (KhudaDad414) [inline .github/workflows/if-docker-pr-testing.yml:36]: @dante381 have you tested these statements in your fork to see if it works?

Comment (dante381) [reply .github/workflows/if-docker-pr-testing.yml:36]: No i didnt. I dont know how to test it.

Comment (KhudaDad414) [reply .github/workflows/if-docker-pr-testing.yml:36]: you have a few options here. either you can use [act](https://github.com/nektos/act) or create another brach, remove the testing logic and replace them with some logging message like `echo "Yay it's working..."` 😆. then you can just change the `allcontributor` and/or `asyncapi-bot` with your own username and check if it actually works as expected.

**Comment by dante381** [comment]
I have tested it using `act` and it raised a few parsing errors due to extra `)` and double quotations `""`.
I have fixed it and there are no other errors.

[Resulting output of the test](https://github.com/asyncapi/.github/files/10941515/test_result.txt)

File: .github/workflows/if-docker-pr-testing.yml [hunk 2]
```diff
@@ -16,7 +16,24 @@ jobs:
     runs-on: ubuntu-latest
     
     steps:
-      - if: "github.event.pull_request.draft == false &&!((github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'ci: update global workflows')) || (github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'chore(release):')) || (github.actor == 'allcontributors' && startsWith(github.event.pull_request.title, 'docs: add')))"
+      - if: >
+          !github.event.pull_request.draft &&!(
+            (github.actor == 'asyncapi-bot' && !(
```
Comment (KhudaDad414) [inline .github/workflows/if-docker-pr-testing.yml:21]:
```suggestion
          !github.event.pull_request.draft && !(
            (github.actor == 'asyncapi-bot' && (
```

File: .github/workflows/if-go-pr-testing.yml [hunk 3]
```diff
@@ -13,7 +13,24 @@ jobs:
     name: Lint Go PR
     runs-on: ubuntu-latest
     steps:
-      - if: "github.event.pull_request.draft == false &&!((github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'ci: update global workflows')) || (github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'chore(release):')) || (github.actor == 'allcontributors' && startsWith(github.event.pull_request.title, 'docs: add')))"
+      - if: >
+          !github.event.pull_request.draft &&!(
+            (github.actor == 'asyncapi-bot' && !(
```
Comment (KhudaDad414) [inline .github/workflows/if-go-pr-testing.yml:18]:
```suggestion
          !github.event.pull_request.draft && !(
            (github.actor == 'asyncapi-bot' && (
```

File: .github/workflows/if-go-pr-testing.yml [hunk 4]
```diff
@@ -43,7 +60,24 @@ jobs:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
     steps:
-      - if: "github.event.pull_request.draft == false &&!((github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'ci: update global workflows')) || (github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'chore(release):')) || (github.actor == 'allcontributors' && startsWith(github.event.pull_request.title, 'docs: add')))"
+      - if: >
+          !github.event.pull_request.draft &&!(
+            (github.actor == 'asyncapi-bot' && !(
```
Comment (KhudaDad414) [inline .github/workflows/if-go-pr-testing.yml:65]:
```suggestion
          !github.event.pull_request.draft && !(
            (github.actor == 'asyncapi-bot' && (
```

File: .github/workflows/if-docker-pr-testing.yml (hunk 2 above)
Comment (dante381) [reply .github/workflows/if-docker-pr-testing.yml]: made the changes

File: .github/workflows/if-go-pr-testing.yml (hunk 3 above)
Comment (dante381) [reply .github/workflows/if-go-pr-testing.yml]: made the changes

File: .github/workflows/if-go-pr-testing.yml (hunk 4 above)
Comment (dante381) [reply .github/workflows/if-go-pr-testing.yml]: made the changes

Comment (KhudaDad414) [reply .github/workflows/if-go-pr-testing.yml]: can you remove that extra `!` as well?

Comment (dante381) [reply .github/workflows/if-go-pr-testing.yml]: Done

File: .github/workflows/if-docker-pr-testing.yml [hunk 5]
```diff
@@ -16,7 +16,24 @@ jobs:
     runs-on: ubuntu-latest
     
     steps:
-      - if: "github.event.pull_request.draft == false &&!((github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'ci: update global workflows')) || (github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'chore(release):')) || (github.actor == 'allcontributors' && startsWith(github.event.pull_request.title, 'docs: add')))"
+      - if: >
+          !(github.event.pull_request.draft &&
+            (github.actor == 'asyncapi-bot' && !(
+              startsWith(github.event.pull_request.title, 'ci: update global workflows') ||
```
Comment (KhudaDad414) [inline .github/workflows/if-docker-pr-testing.yml:22]: the `!` after `'asyncapi-bot' &&` should be removed.

```suggestion
          !github.event.pull_request.draft && !(
            (github.actor == 'asyncapi-bot' && (
              startsWith(github.event.pull_request.title, 'ci: update global workflows') || 
```

File: .github/workflows/if-go-pr-testing.yml [hunk 6]
```diff
@@ -13,7 +13,24 @@ jobs:
     name: Lint Go PR
     runs-on: ubuntu-latest
     steps:
-      - if: "github.event.pull_request.draft == false &&!((github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'ci: update global workflows')) || (github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'chore(release):')) || (github.actor == 'allcontributors' && startsWith(github.event.pull_request.title, 'docs: add')))"
+      - if: >
+          !(github.event.pull_request.draft &&
+            (github.actor == 'asyncapi-bot' && !(
+              startsWith(github.event.pull_request.title, 'ci: update global workflows') ||
```
Comment (KhudaDad414) [inline .github/workflows/if-go-pr-testing.yml:19]:
```suggestion
          !github.event.pull_request.draft && !(
            (github.actor == 'asyncapi-bot' && (
              startsWith(github.event.pull_request.title, 'ci: update global workflows') || 
```

File: .github/workflows/if-go-pr-testing.yml [hunk 7]
```diff
@@ -43,7 +60,24 @@ jobs:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
     steps:
-      - if: "github.event.pull_request.draft == false &&!((github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'ci: update global workflows')) || (github.actor == 'asyncapi-bot' && startsWith(github.event.pull_request.title, 'chore(release):')) || (github.actor == 'allcontributors' && startsWith(github.event.pull_request.title, 'docs: add')))"
+      - if: >
+          !(github.event.pull_request.draft &&
+            (github.actor == 'asyncapi-bot' && !(
+              startsWith(github.event.pull_request.title, 'ci: update global workflows') ||
```
Comment (KhudaDad414) [inline .github/workflows/if-go-pr-testing.yml:66]:
```suggestion
          !github.event.pull_request.draft && !(
            (github.actor == 'asyncapi-bot' && (
              startsWith(github.event.pull_request.title, 'ci: update global workflows') || 
```

**Review by KhudaDad414 — CHANGES_REQUESTED** [review]
PS: you can just click on the `commit suggestion` button and GitHub will apply the suggested changes for ya. 😀

**Review by KhudaDad414 — APPROVED** [review]
LGTM!

**Comment by derberg** [comment]
great job @dante381 !

I checked on one PR that was added after we merged this, and all good, there are no bugs and on [normal PRs](https://github.com/asyncapi/modelina/actions/runs/4417737051/jobs/7743862790) nothing is skipped, but of a certain title is there like https://github.com/asyncapi/modelina/actions/runs/4415237144/jobs/7737945176 the skip takes place. Thanks!!!

**Pull request merged**